src/library/Uno.Material/MaterialResourcesV2.cs

```diff
@@ -15,16 +15,115 @@ namespace Uno.Material
 	/// </summary>
 	public sealed class MaterialResourcesV2 : ResourceDictionary
 	{
+		private const string StylePrefix = "M3Material";
+
 		public MaterialResourcesV2()
 		{
 			Source = new Uri("ms-appx:///Uno.Material/Generated/mergedpages.v2.xaml");
+
+			MapSharedStyleKeys();
 		}
 
 		public bool WithImplicitStyles { set => ExportImplicitStyles(value); }
 
+		private string[] GetSharedStyleKeys()
```

Comment:
replace both collections with this:
```cs
private IEnumerable<(string ResourceKey, string AliasedKey, bool IsDefaultStyle)> GetStyleInfo() {
    var result = new List<(string ResourceKey, string AliasedKey, bool IsDefaultStyle)>();
    Add("M3MaterialCheckBoxStyle", implicit: true);
    Add("AppBarButtonStyle");
    ...
    return result;
    
    void Add(string key, string alias = null, bool implicit = false) =>
        result.Add((key, alias ?? key["M3Material".Length..], implicit));
}
```
and, use this in both ExportImplicitStyles & MapSharedStyleKeys

Reply:
Could probably use an AddRange too:
```
AddRange(defaultStyleKey: "M3MaterialBodyMedium", "M3MaterialBodySmall", "M3MaterialBodyMedium", "M3MaterialBodyLarge", ...);
AddRange(defaultStyleKey: null, "Some", "Control", "ThatDoesntHaveAnImplicityStyle", ...);

void AddRange(string defaultStyleKey, params string[] keys) {
   foreach (var key in keys)
      Add(key, implicit: key == defaultStyleKey);
```